src/read/endianness.jl

```diff
@@ -0,0 +1,68 @@
+is_machine_littleendian() = Base.ENDIAN_BOM == 0x04030201
+is_machine_bigendian() = Base.ENDIAN_BOM == 0x01020304
+
+function is_segy_littleendian(s::IO)
+    bfh = BinaryFileHeader()
+    fh_b2s = fh_byte2sample()
+    
+    old_pos = position(s)
+    seek(s, fh_b2s["DataSampleFormat"])
+    dsf = read(s, typeof(bfh.DataSampleFormat))
+    dsf_bs = bswap(dsf)
+    seek(s, old_pos)
+
+    if (dsf >= 1 && dsf <= 16 && is_machine_littleendian()) ||
+       (dsf_bs >= 1 && dsf_bs <= 16 && is_machine_bigendian())
+        return true
+    else
+        return false
+    end
+end
+
+function is_segy_bigendian(s::IO)
+    bfh = BinaryFileHeader()
+    fh_b2s = fh_byte2sample()
+    
+    old_pos = position(s)
+    seek(s, fh_b2s["DataSampleFormat"])
+    dsf = read(s, typeof(bfh.DataSampleFormat))
+    dsf = bswap(read(s, typeof(bfh.DataSampleFormat)))
```

Comment:
Double read and duplicate var name should be like above. Should probably also be just `is_segy_bigendian(s::IO) = !is_segy_littleendian(s)` or something like that

Those aren't used anywhere what are their purpose?

Reply:
I wasnt sure how whether to use API with is_bigendian or swapbytes_needed. Then I decided to use the second approach and these bug/little functions not needed anymore. Probably Ishould better remove them